app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetMessagesForConversationUseCase.kt

```diff
@@ -50,11 +50,12 @@ class GetMessagesForConversationUseCase @Inject constructor(
         val pagingConfig = PagingConfig(
             pageSize = PAGE_SIZE,
             prefetchDistance = PREFETCH_DISTANCE,
-            initialLoadSize = max(INITIAL_LOAD_SIZE, lastReadIndex + PREFETCH_DISTANCE)
+            initialLoadSize = INITIAL_LOAD_SIZE
         )
         return getMessages(
             conversationId,
-            pagingConfig = pagingConfig
+            pagingConfig = pagingConfig,
+            startingOffset = max(0, lastReadIndex - PREFETCH_DISTANCE)
```

Comment:
Here it is where we really are checking which offset we should start from (either `0` when normally entering the screen, or checking the last read message index position)